### ISSUE TITLE
Change Misleading Website Comparison Table

### ISSUE THREAD

**Issue by presentfactory** [simplex-chat/simplex-chat#5457]
While what SimpleX tries to do seems to be good I am continuously bothered by the comparison table put prominently on the homepage of the website (same table [here](https://github.com/simplex-chat/simplex-chat/blob/stable/docs/SIMPLEX.md#comparison-with-other-protocols) in the docs) as it feels like it is deceptively trying to set SimpleX above other protocols despite many of them being pretty close to what is being offered.

**Possibility of MITM**

Saying Signal, Matrix, or P2P protocols have the "possibility" of MITM while SimpleX is not is kinda nonsense. The footnote says SimpleX prevents this with out of band key exchange/verification, but that's true with any online communication. It is /always/ possible to MITM communications through the internet, it does not matter what protocol it is. The only way to combat it is to do the key exchange via some trusted channel (e.g. out of band like SimpleX and P2P protocols like Tox do) or verify it via a trusted channel that the key exchange was done properly (as is done with Matrix and others do with a hash).

As such I'd personally just remove this row as any good e2e encrypted secure messaging application should be resistant to MITM attacks (as that's kinda the whole point), so it's not really providing any meaningful comparison. If you want to keep it, a new column should be added to compare to things that offer e2e encryption but don't actually give you ways to do the key exchange yourself and/or verify that it was done properly. Examples of that are things that work on more of a "Trust me bro" principle for the key exchange such as Apple's FaceTime, or any proprietary software as you cannot inspect the code easily to ensure the crypto is actually doing as it claims in such cases.

**Single or Centralized network**

This whole row is also fairly confusing as it conflates single and centralized networks together when really they are only a bad thing when combined. Centralization we know is bad as it leads often to a single network you must use which has control over everything, but this does not mean having a single network is a bad thing. For instance in P2P sure with many implementations there's only a single network, but unlike a centralized single network there is no central actual authority over it due to its distributed nature, so it already avoids the problem centralization has.

Additionally, federated networks often form a singular network just the same like P2P networks do, that's kinda the whole point of federation (allowing them to communicate as part of one big network rather than just a number of scattered centralized servers). Sure you can in theory form a new distinct federated network if you really tried to, but you can do the same with P2P too. Nothing about either technology enforces a single network other than that usually being the default configuration to not have the entire userbase fragmented across 20 networks, and this is desired because a single network in such decentralized systems avoids the authority concerns that arise when centralized services do this (that's kinda the whole point of structuring the protocol in this way).

So for this one I'd change the P2P one to "No - distributed" because if federated protocols are going to be considered fine in this regard then P2P should as well (as P2P is the extreme case of any sort of decentralized protocol). I'd also remove the "single network" bit and change it to just "Centralized network" as that is not the problem. The "problem" is centralized authority things like Signal have as it gives a single entity control over things with no alternatives unlike the lack of authority in P2P networks or the ability to create alternative federated instances in federated networks.

**Central component or other network-wide attack**

For this one I really don't understand what it's getting at and once again seems a bit useless of a comparison. Yes there's attacks against every network architecture out there, really a denial of service attack can be applied to any such thing. Usually P2P networks and the like are considered more robust against DDoS in some ways due to how many nodes have to be targeted to actually hurt the network significantly, but of course Sybil attacks and other such things with bad nodes can cause their own sorts of problems. Similarly the lack of central networking can make it harder to mitigate such attacks, vs a big company simply using a DDoS protection service to absorb the traffic and blacklist specific IP ranges across the entire service etc.

All that said, I do not how SimpleX thinks it is "resilient" enough to be classified as a "No" in this category whereas all these other networks are not. SimpleX is not magically immune to attacks from what I understand, one can simply DDoS the message queues and everything will fall apart much like one can do with a federated network's servers.

Finally, nowhere is the claim "P2P networks either have a central authority or the whole network can be compromised" actually explained. If it is referring to Sybil attacks then sure but that's just a denial of service attack and the same can be said about anything else. Is it easier to do than other network types? Maybe in some ways with that sort of approach, but it's not like P2P networks are totally oblivious to the concern either (hence stuff like cryptocurrencies and proof of work/stake systems to mitigate such things due to how important the validity of the consensus is in those cases).

(Originally posted this in the website repo [here](https://github.com/simplex-chat/website/issues/34) but saw that repo has been merged into this one, so moved the issue here).

**Comment by epoberezkin** [comment]
> Saying Signal, Matrix, or P2P protocols have the "possibility" of MITM while SimpleX is not is kinda nonsense.

I agree that with P2P protocols it has to be changed to Maybe or Depends, because some P2P protocols include key material (or hash of key material) into the user address, thus preventing MITM attacks. I will amend that.

With Signal, Matrix, etc., there is indeed a possibility of MITM, because there is no mandatory verification of key exchange. While it is offered as an option, it mitigates the attack, rather than prevents it, and only a small share of users use this verification.

Signal's own argument that a small share of users doing this verification prevents this attack from happening is only correct with regards to large-scale attack that would indeed be detected. But it does not mitigate targeted attacks.

Also don't forget that you have to do this verification out of band every time the app shows "security code changed", and most of the users who verify it on the first connection do not re-verify it on code change.

In addition to that, review this paper that shows how an attack can be performed by adding a user's device without any notification to the user: https://eprint.iacr.org/2021/626.pdf

So overall it seems fair to say that Signal does not prevent MITM attack, it makes some efforts to allow users to mitigate it manually, and optionally, but does not prevent the attack vector via adding another device.

SimpleX prevents MITM attack by including key material in the invitation link, so messaging servers cannot perform MITM attack under any circumstances, without any user action.

> any good E2E encrypted secure messaging application should be resistant to MITM attacks (as that's kinda the whole point)

100% agree with that, but it is not the case with many e2e encrypted applications. Whether that makes them bad is not my call, we're just stating the facts in this table.

> The only way to combat it is to do the key exchange via some trusted channel (e.g. out of band like SimpleX and P2P protocols like Tox do) or verify it via a trusted channel that the key exchange was done properly (as is done with Matrix and others do with a hash).

That is correct, but doing key exchange via trusted channel prevents it, and in case of SimpleX and Tax it happens for 100% of users, non-optionally, and doing verification after key exchange is optional, and it does not prevent MITM attack but mitigates it, and only for a small share of the users who do it every time security code changes.

I agree that it is even worse if the app does not provide facilities to do verification, or is not open-source, but given that most users don't do this verification, it does not make any difference for possibility of MITM attack.

> But unlike a centralized single network there is no central actual authority over it due to its distributed nature, so it already avoids the problem centralization has.

Most p2p networks arrive to the need to have some sort of centralized authority.

I will consider the rest.

**Comment by presentfactory** [comment]
> With Signal, Matrix, etc., there is indeed a possibility of MITM, because there is no mandatory verification of key exchange.

Yes that is fair, having it required ensures no mistakes are made but I would say at least the possibility of verification (especially when something like Matrix informs you that a chat is unverified with a big red symbol) gives it a "Maybe" in that category of being open to MITMs not a "Yes". It's kinda up to the user if they wish to respect that information or not, not that the protocol is incapable of guaranteeing MITM resistance when desired.
That's in contrast to something with a centralized key exchange that you have no way in the client to verify without modifications which is a lot closer to just not having that functionality at all since there's no way to know it's even being done past just trusting some company.

> Most p2p networks arrive to the need to have some sort of centralized authority.

If you're referring to boostrapping into the DHT yes that is centralized in the sense that it's some sort of trusted thing you need to use to get into the DHT to begin with, but it's not actually centralized in terms of its ownership. Tox for instance has a number of boostrap nodes and anyone can just run their own if they feel like the existing ones aren't doing their job properly: https://nodes.tox.chat/
Of course it still requires that say a client knows this list of nodes so you could say whoever controls the client's default configuration has some strong influence there, and of course a lot of nodes use a domain rather than an IP so DNS is involved there too, but it's always possible for people to edit that list themselves in open source software, or connect into the DHT in any way they please really.
I am pretty sure this is standard practice with other DHTs too though since having a centralized entity in control of who can actually join the network is undesirable (like I remember Ethereum's network having something like this as well).
